evm/interchainMultisig.js

```diff
@@ -0,0 +1,563 @@
+    const provider = getDefaultProvider(rpc);
+    const wallet = await getWallet(privateKey, provider, options);
+    await printWalletInfo(wallet, options);
+
+    const multisigContract = new Contract(multisigAddress, IInterchainMultisig.abi, wallet);
+    const gasOptions = await getGasOptions(chain, options, contractName);
+
+    await preExecutionChecks(multisigContract, action, wallet, batchId, calls, signers, weights, threshold, signatures);
+
+    const proof = sortWeightedSignaturesProof(encodeInterchainCallsBatch(batchId, calls), signers, weights, threshold, signatures);
+
+    await multisigContract.executeCalls(formatBytes32String(batchId), calls, proof, gasOptions);
+
+    if (action === 'rotateSigners') {
+        const newSigners = options.newSigners.split(',');
+        const newWeights = options.newWeights.split(',').map(Number);
+        const newThreshold = Number(options.newThreshold);
+
+        contractConfig.signers = newSigners;
+        contractConfig.weights = newWeights;
+        contractConfig.threshold = newThreshold;
+
+        saveConfig(config, options.env);
+    }
+
+    printInfo(`Batch with id ${batchId} successfully executed.`);
+}
+
+async function main(options) {
+    calls = [];
+
+    await mainProcessor(options, processCommand, false);
+
+    printInfo(`Interchain calls`, JSON.stringify(calls, null, 2));
+
+    if (options.offline) {
+        const { batchId, privateKey, yes } = options;
+
+        const wallet = await getWallet(privateKey, null, options);
+        const signature = await signCallsBatch(batchId, calls, wallet);
+
+        if (prompt(`Proceed with signing the calls batch with id ${batchId}?`, yes)) {
+            return;
+        }
+
+        printInfo(`Wallet address`, wallet.address);
+        printInfo(`Signature`, signature);
+    } else {
+        await mainProcessor(options, submitTransactions, false);
+    }
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('multisig').description('Script to manage multisig actions');
+
+    addBaseOptions(program, { address: true });
+
+    program.addOption(new Option('-c, --contractName <contractName>', 'contract name').default('InterchainMultisig'));
+    program.addOption(
+        new Option('--action <action>', 'multisig action')
+            .choices([
+                'signers',
+                'setTokenMintLimits',
+                'transferMintLimiter',
+                'transferMultisig',
+                'rotateSigners',
+                'withdraw',
+                'executeCalls',
+                'setFlowLimits',
+                'voidBatch',
+            ])
+            .makeOptionMandatory(true),
+    );
+    program.addOption(new Option('--offline', 'run script in offline mode'));
+    program.addOption(new Option('--batchId <batchId>', 'The id of the batch to be executed').makeOptionMandatory(true));
+    program.addOption(new Option('--signatures <signatures>', 'Signatures to '));
```

Comment:
```suggestion
    program.addOption(new Option('--signatures <signatures>', 'Signatures to ').env('SIGNATURES'));
```
I would suggest adding the option to input arrays via the ENV file, sometimes it's easier then adding via CLI